src/Core/src/Platform/Windows/MauiWinUIWindow.cs

```diff
@@ -96,5 +112,35 @@ IntPtr NewWindowProc(IntPtr hWnd, uint msg, IntPtr wParam, IntPtr lParam)
 		}
 
 		#endregion
+
+		/// <summary>
+		/// Default the Window Icon to the icon stored in the .exe, if any.
+		/// 
+		/// The Icon can be overriden by callers by calling SetIcon themselves.
+		/// </summary>
+		void SetIcon()
+		{
+			var processPath = Environment.ProcessPath;
+			if (!string.IsNullOrEmpty(processPath))
+			{
+				var index = IntPtr.Zero; // 0 = first icon in resources
+				_windowIcon = ExtractAssociatedIcon(IntPtr.Zero, processPath, ref index);
+				if (_windowIcon != IntPtr.Zero)
+				{
+					var appWindow = AppWindow.GetFromWindowId(Win32Interop.GetWindowIdFromWindow(WindowHandle));
+					if (appWindow is not null)
+					{
+						var iconId = Win32Interop.GetIconIdFromIcon(_windowIcon);
+						appWindow.SetIcon(iconId);
```

Comment:
Can we set the Image in the app title bar to the same Icon? 

https://github.com/dotnet/maui/blob/main/src/Core/src/Platform/Windows/Styles/WindowRootViewStyle.xaml#L26

Reply:
We can set it to a `.png` of the same image, but I don't think we should use this `.ico` file.

I think that should be a follow up/separate change. It will require more plumbing.

Reply:
I logged https://github.com/dotnet/maui/issues/6908 to track this issue.